modules/git/ref.go

```diff
@@ -55,40 +55,57 @@ func (ref *Reference) Commit() (*Commit, error) {
 
 // ShortName returns the short name of the reference
 func (ref *Reference) ShortName() string {
-	if ref == nil {
-		return ""
-	}
-	if strings.HasPrefix(ref.Name, BranchPrefix) {
-		return strings.TrimPrefix(ref.Name, BranchPrefix)
+	return RefName(ref.Name).ShortName()
+}
+
+// RefGroup returns the group type of the reference
+func (ref *Reference) RefGroup() string {
+	return RefName(ref.Name).RefGroup()
+}
+
+// RefName represents a git reference name
+type RefName string
+
+func (ref RefName) IsBranch() bool {
+	return strings.HasPrefix(string(ref), BranchPrefix)
+}
+
+func (ref RefName) IsTag() bool {
```

Comment:
No `IsPull`?

Reply:
We don't know if it's pull or push from the ref name.